test/e2e/network_segmentation.go

```diff
@@ -758,33 +759,42 @@ var _ = Describe("Network Segmentation", func() {
 				podAnno, err := unmarshalPodAnnotation(updatedPod.Annotations, f.Namespace.Name+"/"+userDefinedNetworkName)
 				Expect(err).NotTo(HaveOccurred())
 				framework.Logf("Client pod's annotation for network %s is %v", userDefinedNetworkName, podAnno)
-				Expect(len(podAnno.Routes)).To(Equal(6)) // 3 v4 routes + 3 v6 routes for UDN
+				// number of routes is dependaant on the ip familys supported
+				if isIPv6Supported() && isIPv4Supported() {
+					Expect(len(podAnno.Routes)).To(Equal(6)) // 3 v4 routes + 3 v6 routes for UDN
+				} else {
+					Expect(len(podAnno.Routes)).To(Equal(3)) // either 3 v4 routes or 3 v6 routes for UDN
+				}
 
-				By("asserting the *client* pod can contact the server's v4 IP located outside the cluster")
-				Eventually(func() error {
-					return connectToServer(clientPodConfig, externalIpv4, port)
-				}, 2*time.Minute, 6*time.Second).Should(Succeed())
+				if isIPv4Supported() {
+					By("asserting the *client* pod can contact the server's v4 IP located outside the cluster")
+					Eventually(func() error {
+						return connectToServer(clientPodConfig, externalIpv4, port)
+					}, 2*time.Minute, 6*time.Second).Should(Succeed())
+				}
 
-				By("asserting the *client* pod can contact the server's v6 IP located outside the cluster")
-				Eventually(func() error {
-					return connectToServer(clientPodConfig, externalIpv6, port)
-				}, 2*time.Minute, 6*time.Second).Should(Succeed())
+				if isIPv6Supported() {
+					By("asserting the *client* pod can contact the server's v6 IP located outside the cluster")
+					Eventually(func() error {
+						return connectToServer(clientPodConfig, externalIpv6, port)
+					}, 2*time.Minute, 6*time.Second).Should(Succeed())
+				}
```

Comment:
Please consider export this code to a function, maybe calling it "assertClientExternalConnectivity"

Reply:
done